packages/react-next/src/components/Image/Image.base.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react';
+import { classNamesFunction, getNativeProps, imgProperties } from '../../Utilities';
+import { IImageProps, IImageStyleProps, IImageStyles, ImageCoverStyle, ImageFit, ImageLoadState } from './Image.types';
+import { useMergedRefs } from '@uifabric/react-hooks';
+
+const getClassNames = classNamesFunction<IImageStyleProps, IImageStyles>();
+
+export interface IImageState {
+  loadState?: ImageLoadState;
+}
+
+const SVG_REGEX = /\.svg$/i;
+const KEY_PREFIX = 'fabricImage';
+
+function useLoadState(
+  props: IImageProps,
+  imageElement: React.RefObject<HTMLImageElement>,
+  frameElement: React.RefObject<HTMLDivElement>,
+) {
+  const [loadState, setLoadState] = React.useState<ImageLoadState>(ImageLoadState.notLoaded);
+
+  React.useLayoutEffect(() => {
```

Comment:
To prevent flickers. With `useLayoutEffect`, the state is updated synchronously when the changes are pushed to the DOM, triggering an immediate rerender.